src/mcp_atlassian/server.py

```diff
@@ -1167,6 +1190,54 @@ def format_comment(comment: Any) -> dict:
                     )
                 ]
 
+        elif name == "confluence_attach_content":
+            if not ctx or not ctx.confluence:
+                raise ValueError("Confluence is not configured.")
+
+            # Write operation - check read-only mode
+            if read_only:
+                return [
+                    TextContent(
+                        "Operation 'confluence_attach_content' is not available in read-only mode."
+                    )
+                ]
+
+            content = arguments.get("content")
+            name = arguments.get("name")
+            page_id = arguments.get("page_id")
+
+            if not content or not name or not page_id:
+                return [
+                    TextContent(
+                        type="text",
+                        text="Error: Missing required parameters: content, name, and page_id are required.",
+                    )
+                ]
+
+            page = ctx.confluence.attach_content(
+                content=content, name=name, page_id=page_id
+            )
+            if page is None:
+                logger.error("Error attaching content to Confluence page %s", page_id)
```

Comment:
If `attach_content` is changed to `raise` specific exceptions (like `ApiError`, `RequestException`) instead of returning `None` on failure, this `if page is None:` check should be replaced with corresponding `try...except` blocks to handle those specific errors and format better feedback.